compiler/luci/pass/src/ExpandBroadcastConstPass.cpp

```diff
@@ -65,10 +64,23 @@ luci::CircleConst *create_expanded_constant(luci::CircleConst *node, luci::Circl
   auto const node_data = &node->at<loco::DataType::FLOAT32>(0);
   auto const constant_data = &constant->at<loco::DataType::FLOAT32>(0);
 
-  auto const successor_depth = successor->dim(successor->rank() - 1).value();
-  for (uint32_t d = 0; d < successor_depth; ++d)
-    std::copy(node_data, node_data + node_size, constant_data + d * node_size);
-
+  assert(successor->rank() > 1);
+  if (successor->rank() == 2)
+  {
+    auto const N = successor->dim(successor->rank() - 2).value();
+    auto const D = successor->dim(successor->rank() - 1).value();
+    for (uint32_t n = 0; n < N; ++n)
+      std::fill_n(constant_data + n * D, D, node_data[n]);
+  }
+  if (successor->rank() >= 3)
+  {
+    auto const H = successor->dim(successor->rank() - 3).value();
+    auto const W = successor->dim(successor->rank() - 2).value();
+    auto const D = successor->dim(successor->rank() - 1).value();
+    for (uint32_t h = 0; h < H; ++h)
+      for (uint32_t w = 0; w < W; ++w)
+        std::fill_n(constant_data + h * W * D + w * D, D, node_data[h * W + w]);
+  }
```

Comment:
Does this code work for rank > 3?

Reply:
It seems that N = 1. Could you test with N > 1?

If this is not urgent, please let me continue review after end-to-end draft (https://github.com/Samsung/ONE/issues/14554#issuecomment-2591715173) is uploaded.